src/main/java/org/jboss/threads/ViewExecutor.java

```diff
@@ -307,13 +306,17 @@ void interrupt() {
 
         public void run() {
             boolean removeFromAllWrappersAfterCompletion = true;
+            boolean threadMustBeUnsetUponCompletion = true;
             thread = Thread.currentThread();
             try {
                 for (;;) {
                     lock.lock();
                     try {
                         submittedCount--;
                         runningCount++;
+                        // Interruption may be missed between when a TaskWrapper is submitted
+                        // to the delegate executor, and when the task begins to execute.
+                        if (state == ST_SHUTDOWN_INT_REQ) Thread.currentThread().interrupt();
```

Comment:
Added one more fix when I rebased this change. There's some time which wasn't accounted for between submitting a tack wrapper to the underlying executor and TaskWrapper.run being invoked where we could fail to interrupt the thread.

Reply:
Moving this just after the `thread` field is is set, there's no reason to do this under a lock.